backend/src/state_dump.rs

```diff
@@ -176,3 +177,27 @@ pub async fn dump_state(
 
     Ok(Json(state_dump))
 }
+
+pub async fn public_games(
+    Extension(backend_storage): Extension<HashMapStorage<VersionedGame>>,
+) -> Result<Json<HashMap<String, GameState>>, &'static str> {
+    let mut public_games: HashMap<String, GameState> = HashMap::new();
```

Comment:
I think you only need the list of games, right? This might leak information unintentionally.

Perhaps just `Vec<String>` for the names of the rooms? Or, since I see you're using the number of players below, maybe we should just redact all the sensitive values (e.g. as done in https://github.com/rbtying/shengji/blob/d6fee9e36f503d7a47562418f44c8d4e92f484a2/core/src/game_state/mod.rs#L139)

Reply:
Ah makes sense, done. Created a new struct `PublicGameInfo` to hold the two pieces I know. Also moved the sorting logic to backend to make a bit cleaner